homeassistant/components/homekit/util.py

```diff
@@ -84,12 +88,14 @@
     {
         vol.Optional(CONF_STREAM_ADDRESS): vol.All(ipaddress.ip_address, cv.string),
         vol.Optional(CONF_STREAM_SOURCE): cv.string,
+        vol.Optional(CONF_AUDIO_CODEC, default=DEFAULT_AUDIO_CODEC): cv.string,
         vol.Optional(CONF_SUPPORT_AUDIO, default=False): cv.boolean,
         vol.Optional(CONF_MAX_WIDTH, default=DEFAULT_MAX_WIDTH): cv.positive_int,
         vol.Optional(CONF_MAX_HEIGHT, default=DEFAULT_MAX_HEIGHT): cv.positive_int,
         vol.Optional(CONF_MAX_FPS, default=DEFAULT_MAX_FPS): cv.positive_int,
         vol.Optional(CONF_AUDIO_MAP, default=DEFAULT_AUDIO_MAP): cv.string,
         vol.Optional(CONF_VIDEO_MAP, default=DEFAULT_VIDEO_MAP): cv.string,
+        vol.Optional(CONF_VIDEO_CODEC, default=DEFAULT_VIDEO_CODEC): cv.string,
```

Comment:
Lets go with restricting to valid options here. Same for audio.

```
VALID_VIDEO_CODECS = ['libx264','h264_omx','copy']
```



```suggestion
        vol.Optional(CONF_VIDEO_CODEC, default=DEFAULT_VIDEO_CODEC): vol.In(VALID_VIDEO_CODECS),
```




```
libx264 - No HW acceleration
h264_omx - With HW acceleration
copy - Copy
```

Reply:
Both VALID_VIDEO_CODECS and VALID_AUDIO_CODECS are introduced.
```
VALID_VIDEO_CODECS = ["libx264", "h264_omx", "copy"]
VALID_AUDIO_CODECS = ["libopus", "copy"]
```